pkg/bufferdecoder/net_decoder.go

```diff
@@ -51,21 +51,22 @@ func (decoder *EbpfDecoder) DecodeNetCaptureData(netCaptureData *NetCaptureData)
 	netCaptureData.PacketLength = binary.LittleEndian.Uint32(decoder.buffer[offset : offset+4])
 	netCaptureData.ConfigIfaceIndex = binary.LittleEndian.Uint32(decoder.buffer[offset+4 : offset+8])
 
-	decoder.cursor += int(netCaptureData.GetSizeBytes())
```

Comment:
Can you please explain why did you remove this?

Reply:
so that when calling `DecodeNetPacketEvent`, the packetLength and IfaceIndex could be read again, and appended to the netadataArg.
the other option was to pass this as argument from `processNetEvents()` to `protocolProcessor()` and then to `callProtocolHandler()` and then to `netPacketProtocolHandler()`.

what do you think?